pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -244,6 +259,11 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 		Hostname: resourcenames.K8sServiceFullname(r),
 	}
 
+	// Add the finalizer before creating the ClusterIngress so that we can be sure it gets cleaned up.
+	if err := c.ensureFinalizer(r); err != nil {
+		return err
+	}
+
 	logger.Info("Creating ClusterIngress.")
 	clusterIngress, err := c.reconcileClusterIngress(ctx, r, resources.MakeClusterIngress(r, traffic))
```

Comment:
we should drop owner references for existing cluster ingresses

Reply:
I don't think that they are particularly harmful as-is.  Do you have some reason to think they are?

I actually (after a whole bunch of debugging!) just found a fun and subtle bug (which is why all these e2e tests are exploding), which ironically is caused by my dropping OwnerReferences (our filter on the Owner's Kind).

cc @tcnghia I think this is the reason for the bizarrely long resync earlier, we filtered out all of the ClusterIngress events!  Oops.  I'll push a fix for this shortly.

Reply:
> Do you have some reason to think they are?

Currently `OwnerReferences` being used to across scopes (namespace,cluster) was undefined causing the K8s GC unable to delete the `ClusterIngress`. Breaking foreground cascade deletion and causing deleted objects to be reconciled continuously. 


If the undefined behaviour were to change would it have a material impact on us? Looking at the PR further I don't think so (at this time).